### ISSUE TITLE
interegular not in requirements.txt

### ISSUE THREAD

**Issue by thomasahle** [normal-computing/outlines#197]
I ran `pip install -r requirements-doc.txt`, and then tried to run `examples/parsing.py`.
However I still get
```
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/Users/ahle/outlines/examples/parsing.py", line 18, in <module>
    from outlines.text.parsing import PartialPythonIndenter, copy_parser_state, parse_to_end
  File "/Users/ahle/outlines/outlines/__init__.py", line 4, in <module>
    from outlines.text import prompt
  File "/Users/ahle/outlines/outlines/text/__init__.py", line 2, in <module>
    from .generate import continuation
  File "/Users/ahle/outlines/outlines/text/generate/__init__.py", line 2, in <module>
    from .regex import float, integer, regex
  File "/Users/ahle/outlines/outlines/text/generate/regex.py", line 4, in <module>
    import interegular
ModuleNotFoundError: No module named 'interegular'
```
After `pip install interegular` it is able to run.

**Comment by rlouf** [comment]
You need to install the library `pip install .`. `requirements-doc.txt` only contains the extra requirements to build the documentation (which could be moved to `pyproject.toml` instead).

**Comment by rlouf** [comment]
This was also an issue when doing `pip install .`. Solved by #200.
